mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/PostgresMessageIdMapper.java

```diff
@@ -130,18 +142,44 @@ public Publisher<ComposedMessageIdWithMetaData> findMetadata(MessageId messageId
 
     @Override
     public Flux<MailboxMessage> findReactive(Collection<MessageId> messageIds, MessageMapper.FetchType fetchType) {
-        return mailboxMessageDAO.findMessagesByMessageIds(messageIds.stream().map(PostgresMessageId.class::cast).collect(ImmutableList.toImmutableList()),
-                fetchType)
+        return mailboxMessageDAO.findMessagesByMessageIds(messageIds.stream().map(PostgresMessageId.class::cast).collect(ImmutableList.toImmutableList()), fetchType)
             .flatMap(messageBuilderAndRecord -> {
                 SimpleMailboxMessage.Builder messageBuilder = messageBuilderAndRecord.getLeft();
+                Record record = messageBuilderAndRecord.getRight();
                 if (fetchType == MessageMapper.FetchType.FULL) {
-                    return retrieveFullContent(messageBuilderAndRecord.getRight())
-                        .map(headerAndBodyContent -> messageBuilder.content(headerAndBodyContent).build());
+                    return retrieveFullMessage(messageBuilder, record);
                 }
                 return Mono.just(messageBuilder.build());
             }, ReactorUtils.DEFAULT_CONCURRENCY);
     }
 
+    private Mono<SimpleMailboxMessage> retrieveFullMessage(SimpleMailboxMessage.Builder messageBuilder, Record record) {
```

Comment:
I see a lot of methods were written for adding attachment metadata to `SimpleMailboxMessage.Builder`.
it is similar to what we did in [AttachmentLoader](https://github.com/apache/james-project/blob/3243752f862aa9cf4dff9474878837368f4442c4/mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/AttachmentLoader.java)

Can we reuse it?

Reply:
I remember that AttachmentLoader's method that does not fit this case

Reply:
let me try it a bit

Reply:
caught

- Current `AttachmentLoader` get each by each `MessageAttachmentMetadata` by id 
- Your new code: get list MessageAttachmentMetadata by list id  :+1:

I have a refactor for that  https://github.com/apache/james-project/commit/437550fd687ae4cbfd6986cbb712f2d940453061
can you look it

Reply:
great. tks. could you push it to this PR?

Reply:
I don't have permission, you can fetch my code, and cherry-pick this commit

Reply:
FYI that commit is what is making tests failing on imap mpt tests